### DISCUSSION TITLE
indicating minimality of free resolutions

### DISCUSSION THREAD

**Discussion by wdecker** [oscar-system/Oscar.jl#3099] [Ideas]
@ederc, @jankoboehm: In analogy to `complete::Bool`, should we have a field `minimal::Bool` for `FreeResolution`?

**Comment by ederc** [comment]
Yes, we should have such a flag. We discussed this already some weeks ago, but we wanted to wait until mres and minres are available.

**Comment by wdecker** [comment]
Exccellent, this is useful for code to be delopped for the OSCAR book. We should not wait for `minres`, however, as this is a higher hanging fruit.

**Comment by ederc** [comment]
Question: Should this be a boolean (`true`/`false`)? Isn't it more like `true`/`unknown`?

**Comment by wdecker** [comment]
Yes, correct, this should not be a boolean. I wonder, however, whether we should have true/false/unknown?

**Reply by jankoboehm** [reply]
If we do that, we should have a lazyness triggered by is_minimal to determine that automatically and in particular a function which can determine it. Do we want that? 

We would minimize and sort to check minimality. 

That would be nice, but could also come some time later. For the moment the binary thing would also be ok, since the purpose is just to make minimization the identity in case of true.
